pkg/nar/narinfo/types.go

```diff
@@ -88,6 +89,31 @@ func (n *NarInfo) String() string {
 	return buf.String()
 }
 
+// StoreDir derives in which prefix the store is located, based on the narinfo
+// StorePath field.
+func (n NarInfo) StoreDir() string {
+	i := strings.LastIndex(n.StorePath, "/")
+	return n.StorePath[:i]
+}
+
+// Fingerprint is the digest that will be used with a private key to generate
+// one of the signatures.
+func (n NarInfo) Fingerprint() string {
+	f := "1;" +
+		n.StorePath + ";" +
+		n.NarHash.NixString() + ";" +
+		fmt.Sprintf("%d", n.NarSize) + ";"
```

Comment:
Instead of `fmt.Sprintf` use `strconv.Itoa`, it allocates less and has better performance.